web/src/HUD.tsx

```diff
@@ -413,5 +413,26 @@ export function mergeAppUpdate<K extends keyof HudState>(
     })
   }
 
+  // If no references have changed, don't re-render.
+  //
+  // LogStore handles its own update events, so a change
+  // to LogStore doesn't update its reference.
+  // This makes rendering much, much faster for apps
+  // with lots of logs.
+  if (!hasChange(result, prevState)) {
+    return null
+  }
+
   return result
 }
+
+function hasChange(result: any, prevState: any): boolean {
+  for (let k in result) {
+    let resultV = result[k] as any
+    let prevV = prevState[k] as any
+    if (resultV != prevV) {
```

Comment:
Does strict equality make a difference here? (Just curious.)

Reply:
i don't think it will make a difference, but it certainly doesn't hurt